packages/backend/scripts/createOpStack.ts

```diff
@@ -0,0 +1,214 @@
+// eslint-disable-next-line import/no-internal-modules
+import { RawDiscoveryConfig } from '@l2beat/discovery/dist/discovery/config/RawDiscoveryConfig'
+import { assert, EthereumAddress } from '@l2beat/shared-pure'
+import { exec } from 'child_process'
+import { mkdir, readFile, writeFile } from 'fs/promises'
+import path from 'path'
+import util from 'util'
+
+const execPromise = util.promisify(exec)
+
+interface HelpMode {
+  mode: 'help'
+}
+
+interface OpStackMode {
+  mode: 'opStack'
+  project: string
+  initialAddresses: [EthereumAddress, EthereumAddress]
+}
+
+type Config = HelpMode | OpStackMode
+
+void main().catch((e) => {
+  console.error(e)
+  process.exit(1)
+})
+
+async function main() {
+  const config = parseCliParameters()
+  switch (config.mode) {
+    case 'help':
+      printUsage()
+      break
+    case 'opStack':
+      await createOpStackProject(config.project, config.initialAddresses)
+      break
+  }
+}
+
+async function createOpStackProject(
+  project: string,
+  initialAddresses: EthereumAddress[],
+): Promise<void> {
+  const dir = path.join(__dirname, '..', 'discovery', project, 'ethereum')
+  await mkdir(dir, { recursive: true })
+
+  let projectConfig: RawDiscoveryConfig = configTemplate(
+    project,
+    initialAddresses,
+    names,
+  )
+
+  console.log('Running initial discovery for', project)
+  await writeFile(
+    path.join(dir, 'config.jsonc'),
+    JSON.stringify(projectConfig, null, 2),
+  )
+  console.log('Initial discovery...')
+  await execPromise('yarn discover ethereum ' + project, { cwd: __dirname })
```

Comment:
You can execute a discovery directly through a call to `import { discover } from '@l2beat/discovery'`